### DISCUSSION TITLE
Hello! I'm having an issue with the screenpad that doesn't work!

### DISCUSSION THREAD

**Discussion by parkhongmin-926** [Qonfused/ASUS-ZenBook-Duo-14-UX481-Hackintosh#54] [Post-Install Help]
Hi, thank you for working on the model I have.

I installed macOS Ventura 13.6.9 version normally through the EFI you uploaded.

Most features work!
However, there are some features that don't work.
It is as follows:
1. Bluetooth -> I'll look for it a little more and fix it.
2. Screen Pad and Touch Pad

![image](https://github.com/user-attachments/assets/aff94f21-e473-41ad-94d3-b673c38c8ab4)

In the case of the screen pad, it displays like a bad connection like the picture uploaded, and the touch pad used instead of the mouse does not work at all.

I'm wondering what steps I can take in these cases!

Thank you for reading it.

**Comment by parkhongmin-926** [comment]
If you need any more specific information, please leave a comment it. i will give it immediately

**Comment by Maleficent-Magik** [comment]
Hello! Thanks for your feedback! Indeed, this problem is one of the main drawbacks of this model. The problem of the screenpad “sizzling”.

We've already opened a topic about this problem, so I'll let you have a look here: https://github.com/Qonfused/ASUS-ZenBook-Duo-14-UX481-Hackintosh/issues/4

This problem has been around since August 15, 2022, and ideas have already been proposed without success. Only one person has managed to make it work, but decided to play dumb and not divulge the solution...

**Reply by parkhongmin-926** [reply]
Thanks you for Soooo quick answer. Hmm.. Does it have nothing to do with the version? Ventura, Sonoma, or even the lower version doesn't matter. I have to use the Mac, which is a shame

**Reply by Qonfused** [reply]
This issue is persistent across macOS versions, though there is technically a workaround I discovered here in https://github.com/Qonfused/ASUS-ZenBook-Duo-14-UX481-Hackintosh/issues/4#issuecomment-1987709248. Though you may be able to reproduce this, I wouldn't call it a real solution. So far I have not been able to create a patch that consistently addresses this behavior, so this still needs more investigation.

**Reply by parkhongmin-926** [reply]
And.. I have a question about touchpad. Because ScreenPad Plus and TouchPad are connected in the same way, if I can't fix ScreenPad Plus I can't use TouchPad?

**Reply by Qonfused** [reply]
The trackpad is separate from the screenpad and should still work fine, though the screenpad does have working touchscreen w/ gestures.

The way to think about it is that the screenpad is really two components: a screen, and a (touch) digitizer; the digitizer handles touch input much like a trackpad and operates separately from the screen.

There are some other Asus laptops that have a screenpad instead of a trackpad, so even in that case the trackpad functionality of the screenpad would remain unchanged since the problem exists in the actual display.

**Reply by parkhongmin-926** [reply]
Interesting. Screenpad and touchpad (Trackpad? I don't know what to name it exactly). Neither is working. The problem seems to be getting a little complicated!

**Reply by Qonfused** [reply]
Normally the screenpad's touch digitizer conflicts with the primary display's digitizer (#43), though I have previously observed instances where the trackpad + touchscreen driver (VoodooI2CHID) will die if the two touchscreens conflict and time out.

You can try disabling the screenpad by pressing the middle key above the trackpad, though I'd try restarting or waking the laptop from sleep to see if the issue still persists.
![image](https://github.com/user-attachments/assets/9941b655-9cc0-4e0c-a109-99112ae9d1f9)

**Reply by parkhongmin-926** [reply]
The screen pad itself always turns on automatically when it's turned on. Even if you turn off the screen pad when it's turned on in the middle, the track pad (touch pad) doesn't work constantly. Is there any information I can give you? I don't want to give up the touch pad even if I give up the screen pad!

**Reply by Qonfused** [reply]
Sounds like the trackpad is dead pretty much immediately, which suggests that the GPIO pin the driver is using isn't being used by the trackpad.

For now, I'd disable GPIO pinning by adding the `-vi2c-force-polling` boot arg in your config.plist which essentially falls back into the driver's "safe mode".

You may be able to try another GPIO pin if you modify the pin value used in the [SSDT-GPIO.dsl file](https://github.com/Qonfused/ASUS-ZenBook-Duo-14-UX481-Hackintosh/blob/83efd22fafbbeaf76c5bec7a3fa868ed32c49908/src/ACPI/SSDT-GPIO.dsl#L37). Currently what is listed is `0x55` but you can try any one of `0x17` or `0x34` as well.

**Reply by parkhongmin-926** [reply]
Sorry, I didn't understand the exact meaning of changing from 0x55 to 0x17, 0x34. If you look at the file on the link you gave me, it says 0x0055 on L37, do you mean to change it to 0x0017 or 0x0034?

**Reply by Qonfused** [reply]
Exactly, yes. You can also change it to the shorter form `0x55` if you prefer. When you build the project it should output an updated SSDT-GPIO.aml file in the `dist/EFI/OC/ACPI` folder.

**Reply by parkhongmin-926** [reply]
Sorry, I did "0x0017", "0x0034". but nothing changed. What should i do next step? I simply modified that string 0x0055, I don't know if I did it right, I just simply pasted it with EFI folder and rebooted it, but I don't know if I did it right.

**Reply by Qonfused** [reply]
You'd have to re-build the project or [compile the .dsl file](https://dortania.github.io/Getting-Started-With-ACPI/Manual/compile.html); what you want in the ACPI folder is the output .aml file.


Though as a fallback, you can use the `-vi2c-force-polling` argument but it will probably have very buggy behavior.
